### PR TITLE
Allow wider output

### PR DESCRIPTION
**Pull request by happz**

### CODE REVIEW

File: tmt/utils.py [hunk 1]
```diff
@@ -127,8 +127,9 @@ def unrooted(self) -> 'Path':
 
 # Maximum number of lines of stdout/stderr to show upon errors
 OUTPUT_LINES = 100
-# Default output width
-OUTPUT_WIDTH = 79
+# Default & actual output width
+DEFAULT_OUTPUT_WIDTH = 79
+OUTPUT_WIDTH = int(os.getenv('TMT_TERM_WIDTH', DEFAULT_OUTPUT_WIDTH)) or DEFAULT_OUTPUT_WIDTH
```
Comment (thrix) [inline tmt/utils.py:132]: Could we rather sanity check if somebody does not provide here some crap like `TMT_TERM_WIDTH` is a string not converatble to string?

Comment (happz) [reply tmt/utils.py]: Added in 7efe03da

**Comment by therazix** [comment]
LGTM. However, if `TMT_TERM_WIDTH` is set to a lower number, like 30, the textwrap module throws an exception. Should we consider setting a minimum value for this environment variable, or is it unnecessary in this scenario?"

**Comment by happz** [comment]
> LGTM. However, if `TMT_TERM_WIDTH` is set to a lower number, like 30, the textwrap module throws an exception. Should we consider setting a minimum value for this environment variable, or is it unnecessary in this scenario?"

Can you share the exception? I haven't tried this scenario, I was interested in making use of wider terminal window :)

**Comment by therazix** [comment]
> > LGTM. However, if `TMT_TERM_WIDTH` is set to a lower number, like 30, the textwrap module throws an exception. Should we consider setting a minimum value for this environment variable, or is it unnecessary in this scenario?"
> 
> Can you share the exception? I haven't tried this scenario, I was interested in making use of wider terminal window :)

I tried to run `tmt test show` inside of tmt root directory. `TMT_TERM_WIDTH` was set to 30.

```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/home/fvagner/tmt/tmt/__main__.py", line 26, in <module>
    run_cli()
  File "/home/fvagner/tmt/tmt/__main__.py", line 15, in run_cli
    tmt.cli.main()
  File "/home/fvagner/.virtualenvs/tmt/lib/python3.11/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fvagner/.virtualenvs/tmt/lib/python3.11/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/home/fvagner/.virtualenvs/tmt/lib/python3.11/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fvagner/.virtualenvs/tmt/lib/python3.11/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fvagner/.virtualenvs/tmt/lib/python3.11/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fvagner/.virtualenvs/tmt/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fvagner/.virtualenvs/tmt/lib/python3.11/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fvagner/tmt/tmt/cli.py", line 535, in tests_show
    test.show()
  File "/home/fvagner/tmt/tmt/base.py", line 1222, in show
    echo(tmt.utils.format(key, value))
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fvagner/tmt/tmt/utils.py", line 3289, in format
    formatted_value = format_value(
                      ^^^^^^^^^^^^^
  File "/home/fvagner/tmt/tmt/utils.py", line 3208, in format_value
    formatted_value = _format_value(
                      ^^^^^^^^^^^^^^
  File "/home/fvagner/tmt/tmt/utils.py", line 3179, in _format_value
    return list(formatter(value, window_size, key_color, list_format, wrap))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fvagner/tmt/tmt/utils.py", line 3024, in _format_dict
    v_formatted = _format_value(
                  ^^^^^^^^^^^^^^
  File "/home/fvagner/tmt/tmt/utils.py", line 3179, in _format_value
    return list(formatter(value, window_size, key_color, list_format, wrap))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/fvagner/tmt/tmt/utils.py", line 2994, in _format_str
    yield from textwrap.wrap(stripped_paragraph, width=window_size)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/textwrap.py", line 384, in wrap
    return w.wrap(text)
           ^^^^^^^^^^^^
  File "/usr/lib64/python3.11/textwrap.py", line 359, in wrap
    return self._wrap_chunks(chunks)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/textwrap.py", line 253, in _wrap_chunks
    raise ValueError("invalid width %r (must be > 0)" % self.width)
ValueError: invalid width -13 (must be > 0)

Process finished with exit code 1
```

**Comment by happz** [comment]
@therazix fixed in 648c18bb

File: tmt/utils.py [hunk 2]
```diff
@@ -2906,6 +2917,14 @@ class ListFormat(enum.Enum):
 _FORMAT_VALUE_LIST_ENTRY_INDENT = '  - '
 
 
+def assert_window_size(window_size: Optional[int]) -> None:
+    if window_size is None or window_size > 0:
+        return
+
+    raise GeneralError(
+        f"Allowed widht of terminal exhausted, output cannot fit into {OUTPUT_WIDTH} columns.")
```
Comment (therazix) [inline tmt/utils.py:2925]: Just a small typo here:
```suggestion
        f"Allowed width of terminal exhausted, output cannot fit into {OUTPUT_WIDTH} columns.")
```

File: tests/core/adjust/main.fmf [hunk 3]
```diff
@@ -1,4 +1,4 @@
 summary: Verify test/plan adjustments based on context
 description:
     Perform a basic check using 'tmt test/plan show'.
-require+: [tmt+provision-container]
+#require+: [tmt+provision-container]
```
Comment (therazix) [inline tests/core/adjust/main.fmf:4]: Is this change really relevant to this PR?

**Review by therazix — CHANGES_REQUESTED** [review]

File: tmt/utils.py (hunk 2 above)
Comment (happz) [reply tmt/utils.py]: Fixed in e2d256f8

File: tests/core/adjust/main.fmf (hunk 3 above)
Comment (happz) [reply tests/core/adjust/main.fmf]: Nope, fixed in e2d256f8

**Review by therazix — APPROVED** [review]

File: README.rst [hunk 4]
```diff
@@ -384,6 +384,10 @@ TMT_SHOW_TRACEBACK
     traceback is not printed out. When ``TMT_SHOW_TRACEBACK`` is
     set to any string except ``0``, traceback would be printed out.
 
+TMT_TERM_WIDTH
```
Comment (psss) [inline README.rst:387]: I'm not really sure about the abbreviations. Would you be ok with `TMT_TERMINAL_WIDTH` or `TMT_OUTPUT_WIDTH` as well?

File: tmt/utils.py [hunk 5]
```diff
@@ -2906,6 +2917,14 @@ class ListFormat(enum.Enum):
 _FORMAT_VALUE_LIST_ENTRY_INDENT = '  - '
 
 
+def assert_window_size(window_size: Optional[int]) -> None:
```
Comment (psss) [inline tmt/utils.py:2920]: A short (oneline) docstring explaining motivation for the assert?

Comment (happz) [reply tmt/utils.py:2921]: Added in c6285af6

File: README.rst (hunk 4 above)
Comment (happz) [reply README.rst]: Fixed in c6285af6. Although, `TERM` is quite easy to notice pretty much everywhere in unix lands :)

Comment (psss) [reply README.rst]: Thanks.

**Review by psss — APPROVED** [review]
